### PR TITLE
misc: Replacing list with sequence

### PR DESCRIPTION
**Pull request by ShaolunWang**
Replacing list with sequence in:
- `xdsl/frontend/code_generation.py`
- `xdsl/frontend/python_code_check.py`
- `xdsl/ir.py`
- `xdsl/printer.py`

### CODE REVIEW

File: xdsl/ir.py [hunk 1]
```diff
@@ -437,7 +437,7 @@ def print_parameter(self, printer: Printer) -> None:
 class ParametrizedAttribute(Attribute):
     """An attribute parametrized by other attributes."""
 
-    parameters: list[Attribute] = field(default_factory=list)
+    parameters: Sequence[Attribute] = field(default_factory=list)
```
Comment (superlopuh) [inline xdsl/ir.py:440]: This won't work, we want the class to know the exect type of its stored property

**Review by superlopuh — CHANGES_REQUESTED** [review]
overall good, just need to revert one change

File: xdsl/ir.py (hunk 1 above)
Comment (ShaolunWang) [reply xdsl/ir.py:440]: right i will fix this

**Comment by ShaolunWang** [comment]
For some reason my fork keeps saying that I'm lots of commits behind, despite I kept syncing the forks. Should I revert them?

**Comment by superlopuh** [comment]
hmm that's weird, it should be fixable, but might be easier to start over, your call

**Comment by ShaolunWang** [comment]
I think it's better for me to start over, the weird commits would mess up the history

**Pull request closed without merging**